packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,233 @@
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import {
+  WithAsProp,
+  withSafeTypeForAs,
+  ComponentEventHandler,
+  ShorthandValue,
+  ShorthandCollection,
+} from '../../types'
+import {
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  isFromKeyboard,
+  commonPropTypes,
+  AutoControlledComponent,
+  RenderResultConfig,
+} from '../../lib'
+import { Accessibility } from '../../lib/accessibility/types'
+import Button, { ButtonProps } from '../Button/Button'
+import MenuButton, { MenuButtonProps } from '../MenuButton/MenuButton'
+import { splitButtonBehavior } from '../../lib/accessibility'
+import { MenuProps } from '../Menu/Menu'
+import { MenuItemProps } from '../Menu/MenuItem'
+import { PopupProps } from '../Popup/Popup'
+
+export interface SplitButtonSlotClassNames {
+  toggleButton: string
+}
+
+export interface SplitButtonProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** Shorthand for the main button. */
+  button?: ShorthandValue<ButtonProps>
+
+  /** Initial value for 'open'. */
+  defaultOpen?: boolean
+
+  /** A split button can be disabled. */
+  disabled?: boolean
+
+  /** Shorthand for the menu. */
+  menu?: ShorthandValue<MenuProps> | ShorthandCollection<MenuItemProps>
+
+  /**
+   * Called after user's click on the main button.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onMainButtonClick?: ComponentEventHandler<ButtonProps>
+
+  /**
+   * Called after user's click on a menu item.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onMenuItemClick?: ComponentEventHandler<MenuItemProps>
+
+  /**
+   * Event for request to change 'open' value.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onOpenChange?: ComponentEventHandler<PopupProps>
```

Comment:
```suggestion
  onOpenChange?: ComponentEventHandler<SplitButtonProps>
```

As I see in latest change it should be `SplitButtonProps`